Use version from MANIFEST_URL

dliappis opened the pull request:
This commit derives the version directly from the MANIFEST_URL env var rather than rely on `StackVersion` defined in `Directory.Build.Props`.

Closes https://github.com/elastic/elastic-stack-installers/issues/271

dliappis committed:
Use version from MANIFEST_URL

This commit derives the version directly from the MANIFEST_URL env var
rather than rely on `StackVersion` defined in `Directory.Build.Props`.

Closes https://github.com/elastic/elastic-stack-installers/issues/271

dliappis committed:
Also remove Directory.Build.props

dliappis committed:
Leftovers

dliappis commented:
I believe the test failure is unrelated as described in https://github.com/elastic/elastic-stack-installers/issues/277

alpar-t commented on .buildkite/scripts/dra-publish.sh:
```diff
@@ -19,16 +19,21 @@ VAULT_ROLE_ID=$(echo $DRA_CREDS | jq -r '.role_id')
 fi
 
+# Retrieve version value
+VERSION=$(curl -s --retry 5 --retry-delay 10 "$MANIFEST_URL" | jq -r '.version')
```
`$VERSION` is used above, before it's defined

dliappis replied:
Good catch. Fixed in af501a4676211ac9e15f629159202cfea8b06b01

alpar-t commented on README.md:
```diff
@@ -26,24 +26,6 @@ See [ElastiBuild](https://github.com/elastic/elastic-stack-installers/wiki/Elast
-
-### After a patch release
-
-Update version in `Directory.Build.props` in the branch for the related minor version (ex: https://github.com/elastic/elastic-stack-installers/pull/183).
```
Maybe `Directory.Build.props` should have a place-holder version like `0.0.0` to avoid confusion ?

dliappis replied:
> Maybe `Directory.Build.props` should have a place-holder version like `0.0.0` to avoid confusion ?

Not sure I follow, in this PR we are completely removing the file `Directory.Build.props` as it doesn't seem to be needed anywhere. Additionally all references to manual tasks in the README file.

Do you think we still need to keep the file?

alpar-t replied:
We're all good, my bad. I was looking for a change that removes the property not the entire file, I didn't realise it only holds this one value.

dliappis committed:
PR comment in dra-publish

dliappis committed:
Merge branch 'main' into honor-version-from-stack-url

dliappis commented:
The latest [CI test failures](https://buildkite.com/elastic/elastic-stack-installers/builds/5483#0190503f-1e71-4cc6-9907-7d30aa525934/3842-4530) are related to https://github.com/elastic/elastic-stack-installers/issues/277

dliappis committed:
Allow simulating a trigger

To facilitate testing, we relax the conditionals that control the
pipeline execution.

This commit will be reverted before merging.

dliappis committed:
Fix VERSION bug in publish

Now that we rely on MANIFEST_URL for the $VERSION, we don't need to have
logic whether it contains a -SNAPSHOT suffix; it's already correctly
defined in the manifest.

dliappis committed:
Allow overriding BRANCH

in order to test packaging from PRs, we honor an optional DRA_BRANCH
env var

dliappis committed:
Revert "Fix VERSION bug in publish"

This reverts commit d50acd2e683ceca969d44de3bcf60a9f66b1c338.

dliappis committed:
fix duplication of -SNAPSHOT

dliappis committed:
Revert "Allow overriding BRANCH"

This reverts commit f3bc77545284a1b7c503d66aa7d83093d78e6f29.

dliappis committed:
Revert "Allow simulating a trigger"

This reverts commit 7662c0326cac817728c6a96cc63264fd65115dc5.

dliappis commented:
A test simulating the way that unified release triggers this job has been successful: https://buildkite.com/elastic/elastic-stack-installers/builds/5498#019054b1-ae75-47ae-96df-df84aa5b7f20

I have now reverted https://github.com/elastic/elastic-stack-installers/pull/274/commits/7662c0326cac817728c6a96cc63264fd65115dc5 and https://github.com/elastic/elastic-stack-installers/pull/274/commits/f3bc77545284a1b7c503d66aa7d83093d78e6f29 which was raised to help running the build in the same way as the unified release does.

@alpar-t do you mind taking another look? the only change since you LGTM it last time is two bug fixes:

- https://github.com/elastic/elastic-stack-installers/pull/274/commits/d50acd2e683ceca969d44de3bcf60a9f66b1c338
- https://github.com/elastic/elastic-stack-installers/pull/274/commits/1a1f3099889bf482b93d136e6b20418dc7bf0303

dliappis commented:
Needs to be backported to `8.14`
